src/action.js

```diff
@@ -0,0 +1,182 @@
+            return;
+    }
+}
+
+export async function openPullRequest(output, dependency, ghToken) {
+    const head = `patcher-updates-${dependency}`
+    const title = `[Patcher] Update ${dependency}`
+    const commitMessage = "Update dependencies using Patcher"
+    const commitAuthor = "Grunty"
+    const commitEmail = "grunty@gruntwork.io"
+
+    const body = `
+Updated the \`${dependency}\` dependency using Patcher.
+
+### Update summary
+\`\`\`yaml
+${output}
+\`\`\`
+`
+
+    await exec.exec("git", ["config", "user.name", commitAuthor])
+    await exec.exec("git", ["config", "user.email", commitEmail])
+    await exec.exec("git", ["add", "."])
+    await exec.exec("git", ["commit", "-m", commitMessage])
+    await exec.exec("git", ["checkout", "-b", head])
+
+    const context = github.context;
+
+    await exec.exec("git", ["push", "-f", `https://${ghToken}@github.com/${context.repo.owner}/${context.repo.repo}.git`])
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const repoDetails = await octokit.rest.repos.get({ ...context.repo });
+    const base = repoDetails.data.default_branch;
+    core.debug(`Base branch is ${base}. Opening the PR against it.`)
+
+    try {
+        await octokit.rest.pulls.create({ ...context.repo, title, head, base, body, });
+    } catch (error) {
+        if (error.message?.includes(`A pull request already exists for`)) {
+            core.error(`A pull request for ${head} already exists. The branch was updated.`)
+        } else {
+            throw error
+        }
+    }
+}
+
+export async function downloadRelease(owner, repo, tag, ghToken) {
+    core.info(`Downloading Patcher version ${patcherVersion}`);
+
+    const octokit = new github.getOctokit(ghToken);
+
+    const getReleaseUrl = await octokit.rest.repos.getReleaseByTag({ owner, repo, tag })
+
+    const re = new RegExp(`${osPlatform()}.*amd64`)
+    let asset = getReleaseUrl.data.assets.find(obj => {
+        return re.test(obj.name)
+    })
+
+    const path =  await toolCache.downloadTool(asset.url,
+        undefined,
+        `token ${ghToken}`,
+        {
+            accept: 'application/octet-stream'
+        }
+    );
+
+    core.debug(`Patcher version '${patcherVersion}' has been downloaded at ${path}`);
+    return path
+}
+
+function validateCommand(command) {
+    switch (command) {
+        case "":
+        case "update":
+            return updateCommand;
+        case "report":
+            return reportCommand;
+        default:
+            core.setFailed("Unsupported command - only 'update' and 'report' are supported.");
+            return;
+    }
+}
+
+function updateArgs(updateStrategy, dependency, workingDir) {
+    let args = ["update", noColorFlag, nonInteractiveFlag, skipContainerFlag];
+    if (updateStrategy !== "") {
+        args = args.concat(`--update-strategy=${updateStrategy}`)
+    }
+
+    if (dependency !== "") {
```

Comment:
I don't quite understand what `dependency` is - consider adding a comment to explain this. Perhaps I'm just missing some context but the next like `--target=${dependency}` is confusing to me.

Reply:
Can you understand it from the `action.yml`? https://github.com/gruntwork-io/patcher-action/pull/3/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R13-R14

These are terms from Patcher 😬  I will also add a comment above!

```
Usage: patcher update [options] <WORKING_DIR>

In interactive mode (default), find all your dependencies, and allow you to update them to the latest versions. In
non-interactive mode, update all dependencies to the next safe (non breaking) version.

Options:

   --loglevel value          (default: "info")
   --non-interactive         Run in non-interactive mode. Won't prompt the user for input. Useful for scripts,
                             automation, CI/CD. (default: false)
   --container-image value   Use the specified container image for applying patches. Ignored if
                             --skip-container-runtime is used.
   --skip-container-runtime  Skip using a container runtime for applying patches and perform all operations
                             directly on the host. (default: false)
   --update-strategy value   The update strategy to use in non-interactive mode. Must be one of: [next-safe
                             next-breaking] (default: "next-safe")
   --target value            Limit the update to only a single dependency and all of it's usages. e.g: --target
                             gruntwork-io/terraform-aws-service-catalog/services/ecs-module. Only works with the
                             non-interactive flag.
   --no-color                Disable terminal colors in the update summary written to stdout at the end of the update. Only
                             works with the non-interactive flag. (default: false)
   --help, -h                show help (default: false)
   ```